crates/solver/src/solver/single_order_solver.rs

```diff
@@ -154,48 +212,16 @@ impl Solver for SingleOrderSolver {
 
         let mut settlements = Vec::new();
         let settle = async {
-            let name = self.inner.name();
             while let Some(order) = orders.pop_front() {
-                let Some(fill) = self.fills.order(&order, &auction.external_prices) else {
```

Comment:
Moved mostly unchanged to `solve_single_order()`. Instead of using `continue` or pushing into collections it now uses `SolveResult` to signal the outcome of the request.